core/model/job/itinerary/flow.go

```diff
@@ -0,0 +1,89 @@
+package itinerary
+
+import "fmt"
+
+type StepType int
+
+const (
+	goToLocation StepType = iota
+	pickUpPassengers
+	dropOffPassengers
+)
+
+type step struct {
+	prev     *step
```

Comment:
I'm surprised that this is actually a linked list structure. The structure is a list, of course, but in Python I would just use a `list` or in Java an `ArrayList`. Is it common to implement lists as linked lists in Go?

Reply:
it will be a regular list. The job flow implementation coming up next.